applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -547,6 +617,39 @@ class EigensolverStrategy
     ///@name Protected Operations
     ///@{
 
+    /**
+     * @brief This method assigns settings to member variables
+     * @param ThisParameters Parameters that are assigned to the member variables
+     */
+    void AssignSettings(const Parameters ThisParameters) override
+    {
+        BaseType::AssignSettings(ThisParameters);
+        mMassMatrixDiagonalValue = ThisParameters["eigensolver_diagonal_values"]["mass_matrix_diagonal_value"].GetDouble();
+        mStiffnessMatrixDiagonalValue = ThisParameters["eigensolver_diagonal_values"]["stiffness_matrix_diagonal_value"].GetDouble();
+        mComputeModalDecompostion = ThisParameters["compute_modal_decomposition"].GetBool();
+
+        // Saving the scheme
+        if (ThisParameters["scheme_settings"].Has("name")) {
+            mpScheme =  SchemeFactoryType().Create(ThisParameters["scheme_settings"]);
+        }
+
+        // Setting up the default builder and solver
+        if (ThisParameters["builder_and_solver_settings"].Has("name")) {
+            const std::string& r_name = ThisParameters["builder_and_solver_settings"]["name"].GetString();
+            if (KratosComponents<TBuilderAndSolverType>::Has( r_name )) {
+                // Defining the linear solver
+                auto p_linear_solver = LinearSolverFactoryType().Create(ThisParameters["linear_solver_settings"]);
```

Comment:
this is wrong, an eigensolver is needed here, and this factory doesn't exist yet in cpp